plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/flexible/FlexDeployCommandHandler.java

```diff
@@ -39,14 +47,26 @@ protected DeployPreferencesDialog newDeployPreferencesDialog(Shell shell, IProje
   }
 
   @Override
-  protected StagingDelegate getStagingDelegate(IProject project) {
+  protected StagingDelegate getStagingDelegate(IProject project) throws CoreException {
+    IFacetedProject facetedProject = ProjectFacetsManager.create(project);
+    Preconditions.checkNotNull(facetedProject);
+
     String appYamlPath = new FlexDeployPreferences(project).getAppYamlPath();
     IFile appYaml = project.getFile(appYamlPath);
-    if (!appYaml.exists()) {
-      throw new IllegalStateException("Invalid app.yaml path from project preferences.");
-    }
+    Preconditions.checkState(appYaml.exists(), "Invalid app.yaml path from project preferences.");
```

Comment:
This should be a CoreException, not a state check. State checks are only for internal object state, not external file system state.

Reply:
Got it. Will update it.

Reply:
Guava also offers `Verify`.